### PR TITLE
Update requirements of .NET project to include .NET Core as an option

### PR DESCRIPTION
**Pull request by DanFiedler**
Update requirements of .NET project to include .NET Core as an option and clarify what we are looking for from projects.

### CODE REVIEW

File: challenges.md [hunk 1]
```diff
@@ -159,21 +159,24 @@ Goals: We are interested in how you write well-structured, well-tested code that
...
+* Ensure that your `IHttpModule` or `Middleware` is thread-safe and can correctly handle multiple concurrent requests.
+* Handle multiple encodings and different types of pages.
+* Include a small web application that demonstrates the behavior of your `IHttpModule` (or `Middleware`.) This web application is *not* the focus of the project and you should feel free to use the web application template projects provided by Visual Studio.
 
 In addition, please make sure to do the following:
-* Integrate with a free CI Pipeline to compile, build, test and report in [AppVeyor CI](https://www.appveyor.com/) which is free for any Open Source project in GitHub
-* Automated test coverage
-* Make sure to write an amazing README in your GitHub project that explains what you built, why you built it, how to deploy it up and how to use it. Include the [AppVeyor build badge](https://www.appveyor.com/docs/status-badges/) in your README to show status.
+* Integrate with a free CI Pipeline to compile, build, test and report in [AppVeyor CI](https://www.appveyor.com/) or [Azure DevOps](https://azure.microsoft.com/en-us/services/devops/)
+* Write automated tests
```
Comment (ldemidov) [inline challenges.md]: Includes unit and/or integration tests.

File: challenges.md [hunk 2]
```diff
@@ -159,21 +159,24 @@ Goals: We are interested in how you write well-structured, well-tested code that
...
+* Ensure that your `IHttpModule` or `Middleware` is thread-safe and can correctly handle multiple concurrent requests.
+* Handle multiple encodings and different types of pages.
+* Include a small web application that demonstrates the behavior of your `IHttpModule` (or `Middleware`.) This web application is *not* the focus of the project and you should feel free to use the web application template projects provided by Visual Studio.
 
 In addition, please make sure to do the following:
-* Integrate with a free CI Pipeline to compile, build, test and report in [AppVeyor CI](https://www.appveyor.com/) which is free for any Open Source project in GitHub
-* Automated test coverage
-* Make sure to write an amazing README in your GitHub project that explains what you built, why you built it, how to deploy it up and how to use it. Include the [AppVeyor build badge](https://www.appveyor.com/docs/status-badges/) in your README to show status.
+* Integrate with a free CI Pipeline to compile, build, test and report in [AppVeyor CI](https://www.appveyor.com/) or [Azure DevOps](https://azure.microsoft.com/en-us/services/devops/)
```
Comment (ldemidov) [inline challenges.md]: * If writing an HttpModule, include a script to deploy and remove it from application(s) hosted in IIS

File: challenges.md [hunk 3]
```diff
@@ -159,21 +159,24 @@ Goals: We are interested in how you write well-structured, well-tested code that
...
+Your `IHttpModule` (or `Middleware`) should be designed so that it can be safely added to any ASP.NET (or ASP.NET Core) web application.
+
+Your project must satisfy the following requirements: 
 * Measure the total time spent processing the request. 
-* Measure the time spent by only the IHttpHandler for the request.
 * Measure the size of the response body in bytes. Calculate the minimum, average, and maximum responses seen so far. 
-* Provide a script to install your IHttpModule to all applications hosted on IIS.
+* Add new content to HTML pages to display information gathered by your project.
+* Ensure that your `IHttpModule` or `Middleware` is thread-safe and can correctly handle multiple concurrent requests.
+* Handle multiple encodings and different types of pages.
```
Comment (ldemidov) [inline challenges.md]: + Uses memory and computer resources efficiently and could potentially scale for high-trafficked sites.

File: challenges.md [hunk 4]
```diff
@@ -159,21 +159,24 @@ Goals: We are interested in how you write well-structured, well-tested code that
...
 
 In addition, please make sure to do the following:
-* Integrate with a free CI Pipeline to compile, build, test and report in [AppVeyor CI](https://www.appveyor.com/) which is free for any Open Source project in GitHub
-* Automated test coverage
-* Make sure to write an amazing README in your GitHub project that explains what you built, why you built it, how to deploy it up and how to use it. Include the [AppVeyor build badge](https://www.appveyor.com/docs/status-badges/) in your README to show status.
+* Integrate with a free CI Pipeline to compile, build, test and report in [AppVeyor CI](https://www.appveyor.com/) or [Azure DevOps](https://azure.microsoft.com/en-us/services/devops/)
+* Write automated tests 
+* Include an amazing README in your GitHub project that explains what you built, why you built it, how to deploy it up and how to use it. Include the AppVeyor or Azure DevOps build badge in your README to show status.
```
Comment (ldemidov) [inline challenges.md]: Please write a bit about how you could continue to improve this module if you wanted to make it an actual commercial product.

**Comment by SteveFeldman** [comment]
@DanFiedler do you want this reviewed?

**Comment by DanFiedler** [comment]
Yes please.

Get Outlook for iOS<https://aka.ms/o0ukef>

________________________________

**Pull request merged**